meetings/notes/2024/2024-02-15.md

```diff
@@ -0,0 +1,125 @@
+# JS Structs working session meeting 2024-02-15
+
+> [!NOTE]
+> this was not a TG3 meeting, but a `JS Structs working session` meeting
+
+## Attendees
+
+- SYG
+- MM
+- RBN
+- IID
+- MAH
+- RGN
+- SKI
+- CDA
+
+## Notes
+
+[WasmGC shared memory proposal][wasm] / [shared structs proposal][structs] / [convergence][convergence]
+
+SYG: what is non-negotiable?
+
+MM:
+
+- conventional shared mem multithreading is to time what mem-unsafe languages are to space
+- why they are bad: fault containment
+- in mem-unsafe language, a flaw anywhere can cause problem anywhere else
+  - root causing is hard
+- history of concurrent shared access from diff threads has been a disaster
+  - unworkable paradigm
+  - devs of JS working on different libs, unaware of each other
+    - need defensibility, fault containment
+    - JS is memory-safe, concurrency model in JS is communicating event loops
+      - agents have single thread of control
+      - asyc message passing, postMessage, fetch/xmlhttpreq
+- advocating against:
+
+SYG: I don't think I will convince you to change your philosophical viewpoint
+
+- what is non-negotiable from the proposal?
+
+MM:
+
+- non-negotiable is to introduce conventional shared memory multithreading, fully exposing the hazards of that paradigm to the JS program, such that JS programs now have the burden of maintaing consistney and avoiding deadlock within that paradigm, which is inherently a cooperative path.  concurrent agents all need to cooperate to prevent invariants.  comm event loops is already the concurrency model, and that's where we should focus building on.
```

Comment:
```suggestion
- non-negotiable is to introduce conventional shared memory multithreading, fully exposing the hazards of that paradigm to the JS program, such that JS programs now have the burden of maintaining consistency and avoiding deadlock within that paradigm, which is inherently a cooperative path.  concurrent agents all need to cooperate to prevent invariants.  comm event loops is already the concurrency model, and that's where we should focus building on.
```